### ISSUE TITLE
Problems building TF-Lite Android Benchmark tool

### ISSUE THREAD

**Issue by wezzoid** [tensorflow/tensorflow#36284]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Build machine is an Ubuntu VM with 10GB RAM, Oracle VirtualBox on Win10.
Ubuntu 18.04.3 LTS
kernel 5.3.0-26-generic

- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
Target device - Galaxy Tab S6, Snapdragon 855. Android 9. (and others)

- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.0.1
- Python version: 2.7
- Installed using virtualenv? pip? conda?: no
- Bazel version (if compiling from source): 1.2.1
- GCC/Compiler version (if compiling from source): 7.4.0
- CUDA/cuDNN version: none
- GPU model and memory: none

**Describe the problem**
I have successfully converted Keras models to TF-Lite and run them in a simple Android app.
Now I want to optimise performance, I really need the Benchmark tool to see what operations are being sent to which chips and how long each takes, to see whether they way I've built my models suits the devices.

I can't get the benchmark tool to build. i can't get the camera demo to build either. i have tried various versions of the NDK & SDK and get a variety of errors that look like errors in the code but I'm sure it's just my build environment.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

Build machine is an Ubuntu VM with 10GB RAM, Oracle VirtualBox on Win10.

Ubuntu 18.04.3 LTS
kernel 5.3.0-26-generic

Target device - Galaxy Tab S6, Snapdragon 855. Android 9. 

Bazel version 1.2.1
gcc version 7.4.0

Recent install of Android Studio provides the SDK & NDK as below. (I did try NDK 14 but it didn't seem to help)

Cloned the TF repo. On master branch, commit d75abdf from 28th Jan 2020

Ran ./configure, options:

Please specify the location of python. [Default is /usr/bin/python]: - using default
('which python' shows /usr/bin/python, 'python --version' shows 2.7.17)
Found possible Python library paths:
  /usr/local/lib/python2.7/dist-packages
  /usr/lib/python2.7/dist-packages
Please input the desired Python library path to use.  Default is [/usr/local/lib/python2.7/dist-packages]  - using default
Do you wish to build TensorFlow with XLA JIT support? [Y/n]:  - using default
Do you wish to build TensorFlow with OpenCL SYCL support? [y/N]:  - using default
Do you wish to build TensorFlow with ROCm support? [y/N]:  - using default
Do you wish to build TensorFlow with CUDA support? [y/N]:  - using default
Do you wish to download a fresh release of clang? (Experimental) [y/N]:   - using default
Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is -march=native -Wno-sign-compare]:   - using default
Would you like to interactively configure ./WORKSPACE for Android builds? [y/N]: y
Please specify the home path of the Android NDK to use. [Default is /home/medaphor/Android/Sdk/ndk-bundle]: 
/home/medaphor/Android/Sdk/ndk/21.0.6113669
Please specify the (min) Android NDK API level to use. [Available levels: ['16', '17', '18', '19', '21', '22', '23', '24', '26', '27', '28', '29']] [Default is 21]:  - using default
Please specify the home path of the Android SDK to use. [Default is /home/medaphor/Android/Sdk]:  - using default
Please specify the Android SDK API level to use. [Available levels: ['24', '28', '29']] [Default is 29]: - using default
Please specify an Android build tools version to use. [Available versions: ['24.0.3', '29.0.2']] [Default is 29.0.2]:   - using default

Then following: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/benchmark/android
(looking at the Master branch version of the doc)

1st step is:
(0) Refer to https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/android to edit the WORKSPACE to configure the android NDK/SDK.

Well the configure script does that i believe. Doc is out of date? no action taken.

next step is:
(1) Build for your specific platform, e.g.:
bazel build -c opt \
  --config=android_arm64 \
  --cxxopt='--std=c++11' \
  tensorflow/lite/tools/benchmark/android:benchmark_model

ran that, and get error:

ERROR: /home/medaphor/tensorflow/tensorflow/lite/experimental/delegates/hexagon/BUILD:66:1: C++ compilation of rule '//tensorflow/lite/experimental/delegates/hexagon:hexagon_delegate' failed (Exit 1)
tensorflow/lite/experimental/delegates/hexagon/hexagon_delegate.cc:52:32: error: no member named 'make_unique' in namespace 'std'
    auto hexagon_kernel = std::make_unique<HexagonDelegateKernel>();
                          ~~~~~^
tensorflow/lite/experimental/delegates/hexagon/hexagon_delegate.cc:52:44: error: 'HexagonDelegateKernel' does not refer to a value
    auto hexagon_kernel = std::make_unique<HexagonDelegateKernel>();
                                           ^
./tensorflow/lite/experimental/delegates/hexagon/hexagon_delegate_kernel.h:40:7: note: declared here
class HexagonDelegateKernel {
      ^
tensorflow/lite/experimental/delegates/hexagon/hexagon_delegate.cc:52:67: error: expected expression
    auto hexagon_kernel = std::make_unique<HexagonDelegateKernel>();
                                                                  ^

UPDATE:
I missed this warning:

WARNING: The major revision of the Android NDK referenced by android_ndk_repository rule 'androidndk' is 21. The major revisions supported by Bazel are [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]. Bazel will attempt to treat the NDK as if it was r20. This may cause compilation and linkage problems. Please download a supported NDK version.

So installed NDK 20.0.5594570, 
ran configure again, 
did a bazel clean --expunge,
and tried to build. No more warnings but now get this error:

ERROR: /home/medaphor/tensorflow/tensorflow/lite/experimental/ruy/BUILD:246:1: C++ compilation of rule '//tensorflow/lite/experimental/ruy:detect_arm' failed (Exit 1)
tensorflow/lite/experimental/ruy/detect_arm.cc:59:10: error: use of undeclared identifier 'getauxval'
  return getauxval(AT_HWCAP) & kLocalHwcapAsimddp;
         ^

UPDATE:

changing to:

--cxxopt='--std=c++14'

fixes it. Could you please update the readme's?

**Comment by freedomtan** [comment]
it's fixed in 2f4812d

**Comment by karimnosseir** [comment]
Documentation updated.
